From 40d1da390267e29d936bf0bc8e7eae3dd122d605 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sun, 25 Oct 2026 03:49:47 +0100
Subject: [PATCH 1/1] Remove remote video support

Remove the remote video scope, the online video service is not available
anymore. Remove also its unit test and the libsoup-gnome VAPI file.

Signed-off-by: c4pp4
---
 Makefile.am                 |  2 +-
 configure.ac                |  4 ---
 data/Makefile.am            |  5 +---
 po/POTFILES.in              |  5 ----
 po/POTFILES.skip            |  5 ----
 src/Makefile.am             | 58 -------------------------------------
 tests/unit/Makefile.am      | 26 -----------------
 vapi/Makefile.am            |  5 ----
 vapi/libsoup-gnome-2.4.vapi |  8 -----
 9 files changed, 2 insertions(+), 116 deletions(-)
 delete mode 100644 vapi/Makefile.am
 delete mode 100644 vapi/libsoup-gnome-2.4.vapi

diff --git a/Makefile.am b/Makefile.am
index 7f0e7b0..8f8cd28 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = src po tests/unit vapi data
+SUBDIRS = src po tests/unit data
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-localinstall
 
diff --git a/configure.ac b/configure.ac
index 6df84f3..5285c71 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,8 +73,6 @@ PKG_CHECK_MODULES(SCOPE_DAEMON,
                   gio-unix-2.0 >= $GLIB_REQUIRED
                   dee-1.0 >= 1.0.7
                   gee-0.8
-                  libsoup-gnome-2.4
-                  json-glib-1.0
                   zeitgeist-1.0 >= 0.3.8
                   unity >= 7.0.0
                   unity-extras >= 6.91.11
@@ -175,13 +173,11 @@ AC_CONFIG_FILES([
   Makefile
   data/Makefile
   data/local.scope.in
-  data/remote.scope.in
   src/Makefile
   po/Makefile.in
   src/config.vala
   tests/unit/config-tests.vala
   tests/unit/Makefile
-  vapi/Makefile
 ])
 AC_OUTPUT
 
diff --git a/data/Makefile.am b/data/Makefile.am
index e10dbd9..4398abf 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -1,6 +1,5 @@
 dbus_servicesdir = $(DBUSSERVICEDIR)
 service_in_files =  \
-  unity-scope-video-remote.service.in \
   unity-scope-video-local.service.in \
   $(NULL)
 
@@ -9,7 +8,7 @@ dbus_services_DATA = $(service_in_files:.service.in=.service)
 %.service: %.service.in
 	$(AM_V_GEN)sed -e "s|\@pkglibexecdir\@|$(pkglibexecdir)|" $< > $@
 
-scope_in_files = local.scope.in remote.scope.in
+scope_in_files = local.scope.in
 scopedir = $(SCOPESDIR)/video
 scope_DATA = $(scope_in_files:.scope.in=.scope)
 
@@ -21,8 +20,6 @@ EXTRA_DIST = \
   $(NULL)
 
 CLEANFILES = \
-  unity-scope-video-remote.service \
   unity-lens-video.service \
   local.scope \
-  remote.scope \
   $(NULL)
diff --git a/po/POTFILES.in b/po/POTFILES.in
index 283af11..7374f1d 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -1,10 +1,5 @@
 [encoding: UTF-8]
 src/daemon.vala
 src/main.vala
-src/remote-scope.vala
-src/remote-video-main.vala
 src/scope.vala
-src/ubuntu-video-search.vala
-tests/unit/test-ubuntu-video-search.vala
 [type: gettext/ini]data/local.scope.in.in
-[type: gettext/ini]data/remote.scope.in.in
diff --git a/po/POTFILES.skip b/po/POTFILES.skip
index bc936b7..9355242 100644
--- a/po/POTFILES.skip
+++ b/po/POTFILES.skip
@@ -1,7 +1,2 @@
 src/daemon.c
 src/scope.c
-src/remote-scope.c
-src/ubuntu-video-search.c
-tests/unit/test-ubuntu-video-search.c
-tests/unit/ubuntu-video-search.c
-tests/unit/remote-scope.c
diff --git a/src/Makefile.am b/src/Makefile.am
index 24eb16d..3cd6697 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -11,7 +11,6 @@ AM_LDFLAGS = $(COVERAGE_LDFLAGS)
 
 pkglibexec_PROGRAMS = \
 	unity-video-lens-daemon \
-	unity-scope-video-remote \
 	$(NULL)
 
 unity_video_lens_daemon_CPPFLAGS = \
@@ -24,16 +23,6 @@ unity_video_lens_daemon_CPPFLAGS = \
   -I$(srcdir) \
   $(NULL)
 
-unity_scope_video_remote_CPPFLAGS = \
-  -DDATADIR=\"$(DATADIR)\" \
-  -DPKGDATADIR=\"$(PKGDATADIR)\" \
-  -DGETTEXT_PACKAGE=\"$(GETTEXT_PACKAGE)\" \
-  -DG_LOG_DOMAIN=\"unity-scope-video-remote\" \
-  $(SCOPE_DAEMON_CFLAGS) \
-  $(MAINTAINER_CFLAGS) \
-  -I$(srcdir) \
-  $(NULL)
-
 unity_video_lens_daemon_VALAFLAGS = \
   -C \
   --pkg dee-1.0 \
@@ -45,39 +34,14 @@ unity_video_lens_daemon_VALAFLAGS = \
   --pkg gee-0.8 \
   --pkg zeitgeist-1.0 \
   --vapidir $(srcdir) \
-  --vapidir $(top_srcdir)/vapi \
   --target-glib=2.26 \
   $(MAINTAINER_VALAFLAGS) \
   $(NULL)
 
-unity_scope_video_remote_VALAFLAGS = \
-  -C \
-  --pkg dee-1.0 \
-  --pkg unity \
-  --pkg unity-extras \
-  --pkg gio-2.0 \
-  --pkg gio-unix-2.0 \
-  --pkg glib-2.0 \
-  --pkg gee-0.8 \
-  --pkg zeitgeist-1.0 \
-  --pkg libsoup-gnome-2.4 \
-  --pkg libsoup-2.4 \
-  --pkg json-glib-1.0 \
-  --vapidir $(srcdir) \
-  --vapidir $(top_srcdir)/vapi \
-  --target-glib=2.26 \
-  $(MAINTAINER_VALAFLAGS) \
-  $(NULL)
-
-
 unity_video_lens_daemon_LDADD = \
   $(SCOPE_DAEMON_LIBS) \
   $(NULL)
 
-unity_scope_video_remote_LDADD = \
-  $(SCOPE_DAEMON_LIBS) \
-  $(NULL)
-
 unity_video_lens_daemon_VALASOURCES = \
   blacklist-tracker.vala \
   daemon.vala \
@@ -90,46 +54,24 @@ unity_video_lens_daemon_VALASOURCES = \
   video-file.vala \
   $(NULL)
 
-unity_scope_video_remote_VALASOURCES = \
-  config.vala \
-  remote-scope-globals.vala \
-  remote-video-main.vala \
-  remote-scope.vala \
-  ubuntu-video-search.vala \
-  utils.vala \
-  video-file.vala \
-  $(NULL)
-
 unity_video_lens_daemon_SOURCES = \
   $(unity_video_lens_daemon_VALASOURCES:.vala=.c) \
   $(NULL)
 
-unity_scope_video_remote_SOURCES = \
-  $(unity_scope_video_remote_VALASOURCES:.vala=.c) \
-  $(NULL)
-
 BUILT_SOURCES += \
   unity_video_lens_daemon.vala.stamp \
-  unity_scope_video_remote.vala.stamp \
   $(NULL)
 
 EXTRA_DIST += \
   unity_video_lens_daemon.vala.stamp \
-  unity_scope_video_remote.vala.stamp \
   $(unity_video_lens_daemon_VALASOURCES) \
-  $(unity_scope_video_remote_VALASOURCES) \
   $(NULL)
 
 unity_video_lens_daemon.vala.stamp: $(unity_video_lens_daemon_VALASOURCES)
 	$(AM_V_GEN) $(VALAC) $(unity_video_lens_daemon_VALAFLAGS) $^
 	@touch $@
 
-unity_scope_video_remote.vala.stamp: $(unity_scope_video_remote_VALASOURCES)
-	$(AM_V_GEN) $(VALAC) $(unity_scope_video_remote_VALAFLAGS) $^
-	@touch $@
-
 CLEANFILES += \
   *.stamp \
   $(unity_video_lens_daemon_VALASOURCES:.vala=.c) \
-  $(unity_scope_video_remote_VALASOURCES:.vala=.c) \
   $(NULL)
diff --git a/tests/unit/Makefile.am b/tests/unit/Makefile.am
index f0fadd8..682b9fc 100644
--- a/tests/unit/Makefile.am
+++ b/tests/unit/Makefile.am
@@ -3,7 +3,6 @@ include $(top_srcdir)/Makefile.decl
 check_PROGRAMS = \
 	test-locate \
 	test-utils \
-        test-ubuntu-video-search \
 	$(NULL)
 
 TEST_PROGS += $(check_PROGRAMS)
@@ -14,14 +13,10 @@ AM_VALAFLAGS = \
 	  --pkg unity-extras \
 	  --pkg gio-2.0 \
 	  --pkg gio-unix-2.0 \
-	  --pkg json-glib-1.0 \
 	  --pkg glib-2.0 \
 	  --pkg gee-0.8 \
 	  --pkg zeitgeist-1.0 \
-	  --pkg libsoup-gnome-2.4 \
-	  --pkg libsoup-2.4 \
 	  --vapidir $(srcdir) \
-	  --vapidir $(top_srcdir)/vapi \
 	  --target-glib=2.26 \
 	  $(MAINTAINER_VALAFLAGS) \
 	  $(NULL)
@@ -58,22 +53,10 @@ test_utils_VALASOURCES = \
 	$(top_srcdir)/src/utils.vala \
 	$(NULL)
 
-test_ubuntu_video_search_VALASOURCES = \
-	test-ubuntu-video-search.vala \
-	config-tests.vala \
-	$(top_srcdir)/src/remote-scope-globals.vala \
-	$(top_srcdir)/src/ubuntu-video-search.vala \
-	$(top_srcdir)/src/video-file.vala \
-	$(top_srcdir)/src/remote-scope.vala \
-	$(top_srcdir)/src/utils.vala \
-	$(NULL)
-
 nodist_test_locate_SOURCES = $(test_locate_VALASOURCES:.vala=.c)
 
 nodist_test_utils_SOURCES = $(test_utils_VALASOURCES:.vala=.c)
 
-nodist_test_ubuntu_video_search_SOURCES = $(test_ubuntu_video_search_VALASOURCES:.vala=.c)
-
 CLEANFILES = *.stamp \
 	  *.c \
 	  $(NULL)
@@ -81,10 +64,6 @@ CLEANFILES = *.stamp \
 EXTRA_DIST = \
 	$(test_locate_VALASOURCES) \
 	$(test_utils_VALASOURCES) \
-	$(test_ubuntu_video_search_VALASOURCES) \
-	data/videosearch_input1.txt \
-	data/videosearch_input2.txt \
-	data/videosearch_details1.txt \
         data/video1.avi \
         data/video2.avi \
         data/video3.avi \
@@ -96,7 +75,6 @@ EXTRA_DIST = \
 BUILT_SOURCES = \
 	test-locate.vala.stamp \
 	test-utils.vala.stamp \
-	test-ubuntu-video-search.vala.stamp \
 	$(NULL)
 
 test-locate.vala.stamp: $(test_locate_VALASOURCES)
@@ -107,10 +85,6 @@ test-utils.vala.stamp: $(test_utils_VALASOURCES)
 	$(AM_V_GEN)$(VALAC) -C $(AM_VALAFLAGS) $(VALAFLAGS) $^
 	@touch $@
 
-test-ubuntu-video-search.vala.stamp: $(test_ubuntu_video_search_VALASOURCES)
-	$(AM_V_GEN)$(VALAC) -C $(AM_VALAFLAGS) $(VALAFLAGS) $^
-	@touch $@
-
 # START HEADLESS TESTS
 if ENABLE_HEADLESS_TESTS
 test-headless:
diff --git a/vapi/Makefile.am b/vapi/Makefile.am
deleted file mode 100644
index f3a4d38..0000000
--- a/vapi/Makefile.am
+++ /dev/null
@@ -1,5 +0,0 @@
-NULL =
-BUILT_SOURCES =
-CLEANFILES =
-EXTRA_DIST = libsoup-gnome-2.4.vapi
-
diff --git a/vapi/libsoup-gnome-2.4.vapi b/vapi/libsoup-gnome-2.4.vapi
deleted file mode 100644
index 3af2562..0000000
--- a/vapi/libsoup-gnome-2.4.vapi
+++ /dev/null
@@ -1,8 +0,0 @@
-[CCode (cprefix = "Soup", gir_namespace = "SoupGNOME", gir_version = "2.4", lower_case_cprefix = "soup_")]
-namespace SoupGNOME {
-	[CCode (cheader_filename = "libsoup/soup-gnome.h", type_id = "soup_proxy_resolver_gnome_get_type ()")]
-	public class ProxyResolverGNOME : Soup.ProxyResolverDefault, Soup.ProxyURIResolver, Soup.SessionFeature {
-		[CCode (has_construct_function = false)]
-		protected ProxyResolverGNOME ();
-	}
-}
-- 
2.55.0

